src/jit/emitxarch.cpp

```diff
@@ -4812,6 +4819,23 @@ void emitter::emitIns_SIMD_R_R_R(instruction ins, regNumber reg, regNumber reg1,
         emitIns_R_R(ins, emitTypeSize(simdtype), reg, reg2);
     }
 }
+
+void emitter::emitIns_SIMD_R_R_R_I(
+    instruction ins, regNumber reg, regNumber reg1, regNumber reg2, int ival, var_types simdtype)
+{
+    if (UseVEXEncoding())
+    {
+        emitIns_R_R_R_I(ins, emitTypeSize(simdtype), reg, reg1, reg2, ival);
+    }
+    else
+    {
+        if (reg1 != reg)
+        {
+            emitIns_R_R(INS_movaps, emitTypeSize(simdtype), reg, reg1);
+        }
+        emitIns_R_R_I(ins, emitTypeSize(simdtype), reg, reg2, ival);
```

Comment:
Why `reg, reg2` and not `reg, reg1`?

Reply:
Because `reg` is `reg1` at this point. `ins reg1, reg1, ival` wouldn't do what we need.